src/transformers/models/siglip/convert_siglip_to_hf.py

```diff
@@ -263,37 +273,54 @@ def convert_siglip_checkpoint(model_name, pytorch_dump_folder_path, verify_logit
 
     # get checkpoint
     checkpoint = model_name_to_checkpoint[model_name]
-
+    if (config.vision_config.image_size==256 and config.text_config.vocab_size==250000 and config.vision_config.patch_size==16):
+        siglip_sovit_i18_256 = True
+    
     # get vocab file
     if "i18n" in model_name:
-        vocab_file = "/Users/nielsrogge/Documents/SigLIP/multilingual_vocab/sentencepiece.model"
+        vocab_file = "/Users/mervenoyan/Desktop/siglips/mc4/sentencepiece.model"
     else:
-        vocab_file = "/Users/nielsrogge/Documents/SigLIP/english_vocab/sentencepiece.model"
+        vocab_file = "/Users/mervenoyan/Desktop/siglips/c4_en/sentencepiece.model"
 
     # load original state dict
     data = load(checkpoint)
     state_dict = flatten_nested_dict(data)
 
-    # remove and rename some keys
+    if siglip_sovit_i18_256:  # make state dict compatible with rest of the SiglIPs, add param/ prefix and encoderblock index
```

Comment:
this model has a single `encoderblock` and not `encoderblock_i` for all hidden layers in vision tower like others so I modified the state dict to be compatible with the rest of the modelling code  @NielsRogge